library/src/main/java/com/yausername/youtubedl_android/YoutubeDL.java

```
@@ -143,16 +147,30 @@ public VideoInfo getInfo(YoutubeDLRequest request) throws YoutubeDLException, In
     }
 
     public YoutubeDLResponse execute(YoutubeDLRequest request) throws YoutubeDLException, InterruptedException {
-        return execute(request, null);
+        return execute(request, null, null);
     }
 
     private boolean ignoreErrors(YoutubeDLRequest request, String out) {
         return request.hasOption("--dump-json") && !out.isEmpty() && request.hasOption("--ignore-errors");
     }
 
     public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadProgressCallback callback) throws YoutubeDLException, InterruptedException {
-        assertInit();
+        return execute(request, callback, null);
+    }
+
+    public boolean destroyProcessById(@NonNull String id) {
+        if (id2Process.containsKey(id)) {
+            id2Process.get(id).destroy();
+            return true;
+        }
+        else
+            return false;
+    }
 
+    public YoutubeDLResponse execute(YoutubeDLRequest request, @Nullable DownloadProgressCallback callback, @Nullable String processId) throws YoutubeDLException, InterruptedException {
+        assertInit();
+        if (processId != null && id2Process.containsKey(processId))
+            return null;
```

Maybe we can throw an error here